src/main/scala/scorex/core/network/peer/PeerManager.scala

```diff
@@ -83,70 +87,68 @@ class PeerManager(settings: ScorexSettings, timeProvider: NetworkTimeProvider) e
   private def isSelf(address: InetSocketAddress, declaredAddress: Option[InetSocketAddress]): Boolean = {
     // TODO: should the peer really be considered the same as self iff one of the following conditions hold?? Check carefully.
     settings.network.bindAddress == address ||
-    settings.network.declaredAddress.exists(da => declaredAddress.contains(da)) ||
-    declaredAddress.contains(settings.network.bindAddress) ||
-    settings.network.declaredAddress.contains(address)
+      settings.network.declaredAddress.exists(da => declaredAddress.contains(da)) ||
+      declaredAddress.contains(settings.network.bindAddress) ||
+      settings.network.declaredAddress.contains(address)
   }
 
+  private var lastIdUsed = 0
 
   private def peerCycle: Receive = {
-    case Connected(newPeer@ConnectedPeer(remote, _)) =>
-      if (peerDatabase.isBlacklisted(newPeer.socketAddress)) {
+    case Connecting(remote, direction) =>
```

Comment:
A minor style remark: a message with name ending with "-ing" seems strange. Maybe we should try to stick to past participle and imperative names (e.g. "DoSomething", telling an actor to do something, and "Done", informing an actor that something was done.)

Reply:
Fixed, it is DoConnecting now